src/main/java/com/meli/hseixas/listaspartidasfutebol/service/PartidaService.java

```diff
@@ -182,4 +176,38 @@ public boolean validarEstadioPorDia(PartidaDto partidaDto) {
         }
         return true;
     }
+
+    public boolean validarIntervaloDeDiasPorClube(PartidaDto partidaDto) {
+        List<Partida> partidasDosClubes = obterPartidasDosClubes(partidaDto);
+        for (Partida partida : partidasDosClubes) {
+            if (intervaloMenorQueDoisDias(partidaDto, partida)){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private List<Partida> obterPartidasDosClubes (PartidaDto partidaDto){
+        List<Partida> todasAsPartidas = partidaRepository.findAll();
+        List<Partida> partidasDosClubes = new ArrayList<>();
+        for (Partida partida : todasAsPartidas){
+            if (partidaEnvolvendoClubes(partida, partidaDto)){
+                partidasDosClubes.add(partida);
+            }
+        }
+        return partidasDosClubes;
+    }
+
+    private boolean partidaEnvolvendoClubes(Partida partida, PartidaDto partidaDto){
+        return partida.getClubeMandante().equalsIgnoreCase(partidaDto.getClubeMandante()) ||
+                partida.getClubeMandante().equalsIgnoreCase(partidaDto.getClubeVisitante()) ||
+                partida.getClubeVisitante().equalsIgnoreCase(partidaDto.getClubeMandante()) ||
+                partida.getClubeVisitante().equalsIgnoreCase(partidaDto.getClubeVisitante());
+    }
+
+    private boolean intervaloMenorQueDoisDias (PartidaDto partidaDto, Partida partida){
```

Comment:
não é legal colocar os valores de checagem no nome do método, pois se vc mudar depois a checagem, terá que alterar o nome do método em todos os lugares

também como é uma checagem se É algo, deve começar com `is`. 

ex.: `isIntervaloValidoEntrePartidas`

Reply:
Feito!